### PR TITLE
feat: apply multi type of chaos nodes with code generation

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

Make workflow supports all the type of chaos

### What is changed and how does it work?

- new method which mapping `templateType` and the CRD Kind, reuse the `chaosKindMap`
- spawn the chaos object from `EmbedChaos` by `templateType`
- tests on dependency of `chaosKindMap`
- appending the example workflow with stress-chaos

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by STRRL** [comment]
I think this PR is ready for review. 😁
/cc @chaos-mesh/committers 
/cc @chaos-mesh/maintainers

File: helm/chaos-mesh/values.yaml [hunk 1]
```diff
@@ -249,7 +249,7 @@ dashboard:
         tlsSecret: dashboard.local-tls
 
 dnsServer:
-  create: false
+  create: true
```
Comment (Colstuwjx) [inline helm/chaos-mesh/values.yaml:252]: It changed the default behavior of the chart installation. Why do we need to update this line inside this PR?

File: pkg/workflow/controllers/chaos_node_reconciler.go [hunk 2]
```diff
@@ -51,51 +49,32 @@ func (it *ChaosNodeReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
+	if !v1alpha1.IsChoasTemplateType(node.Spec.Type) {
+		return reconcile.Result{}, nil
```
Comment (Colstuwjx) [inline pkg/workflow/controllers/chaos_node_reconciler.go:53]: Should we return an error here?

**Review by Colstuwjx — COMMENTED** [review]
rest LGTM.

File: pkg/workflow/controllers/chaos_node_reconciler.go (hunk 2 above)
Comment (STRRL) [reply pkg/workflow/controllers/chaos_node_reconciler.go:53]: No, this reconciler would receive events about all workflownode, event type is not `Type<Some>Chaos`, and the `v1alpha1.IsChoasTemplateType()` plays as a filter.

**Comment by STRRL** [comment]
Hi, as the GcpChaos has been merged https://github.com/chaos-mesh/chaos-mesh/pull/1526, this PR updates the definition of gcp chaos, so it need review again.

/cc @chaos-mesh/committers @chaos-mesh/reviewers

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
PTAL @AsterNighT @Colstuwjx

**Pull request merged**